integration-tests/main/src/test/java/io/quarkus/it/main/QuarkusTestCallbacksTestCase.java

```diff
@@ -23,6 +25,11 @@ public void testTestMethodHasAnnotation() {
 
     }
 
+    @Test
+    public void testBeforeClass() {
+        assertEquals(1, SimpleAnnotationCheckerBeforeClassCallback.count.get());
```

Comment:
👍 for adding a test for it! AFAICS it would be more correct if the test would make sure that `testBeforeClass()` is really called last via `@Order`. 
https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-execution-order

Reply:
Not sure what you mean here

Reply:
Without `@Order` there is no guarantee that this new method is called *after* the existing one.
If it is called as the first method and there is (for instance) a bug that the new callback is called for every method, then there will be a false positive.

Reply:
Not really, because the callback is executed for each test, not just this one. 
So this test basically ensure that the callback is passed the proper parameter.

Reply:
The callback shall run _once_ for `QuarkusTestCallbacksTestCase`, but _not again_ for each of the methods of this test class, right?
So, if I'm not missing something, in case there is a bug and the callback is invoked too often (e.g. for each test method), the test will _not_ detect this when `testBeforeClass()` is called before `testTestMethodHasAnnotation()`.

Reply:
Yeah, that's the case. To be honest I don't think it's something that could happen, but on the other hand it's not a problem for me to add it.

Reply:
Up to you! Just something I ran into a couple of times, especially on different operating systems (but mostly with JUnit 4).
And in some older major JDK update the order of methods gathered via reflection changed (in a non-deterministic way) which had some nasty effects for some projects that (partially unknowingly) relied on some order.

Reply:
Fixed :)